### PR TITLE
I983 custom honors solved count

### LINKED ISSUE #983: Add contingency for High Honors and Honors ranking groups
**Is your feature request related to a problem?** 
This feature may help address a possible "last minute" issue with rankings.
...

**Feature Description**:   

It is a distinct possibility that given the new "_Highest Honors_", "_High Honors_" and "_Honors_" ranking groups for WF type contests (#978), that the criteria for where the various groups start in the rankings could change.  As it stands now, it is possible that the "_High Honors_" (or even "_Honors_") groups may have no teams in them.  For example, medalist ranks 1-12 have 8 problems solved, and no one has 7 solved, then the "_High Honors_" group would not have any teams.  This feature request adds the ability to manually specify the number of problems needed for "High Honors" and "Honors".

In the example above, it may be desired to move any team that has 6 solved up to the "_High Honors_" group, and possibly shifting the "_Honors_" group up as well.
...

**Have you considered other ways to accomplish the same thing?** 
There would be no way to do this.  This is more of an "ICPC HQ" defensive programming feature.
...

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?** 
The code should be able to work either way.  Perhaps on the "**Finalize**" tab, where it provides a mechanism to configure the Gold/Silver/Bronze medals, we could add a couple more boxes that say **# of problems for High Honors**, and **# of problems for Honors**.  Then, those could be entered by hand into PC^2 before we generate the `results.tsv`.  If those boxes are empty (or 0?) then the code will follow the rules as they are currently written (again, see #978).
...

**Additional context**: 
...

### PR DESCRIPTION
**Pull request by SamanwaySadhu**
### Description of what the PR does
Add option to customize the minimum number of problems solved to get Highest Honors, High Honors or Honors through the Finalize Pane in case we're using WF style rankings.

### Issue which the PR addresses
Fixes #983 

### Environment in which the PR was developed (OS, IDE, Java version, etc.)
Windows 10, Eclipse 2021-12 R, JDK 8u381 (1.8), C++ 13.2.0

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)
- Run the Ant script build.xml in the pc2v9 project.
- Start a PC2 server, loading some prefinalized contest `profiles` folder and `profiles.properties` files.
- Start a PC2 Admin.
- Go to `Run Contest` and `Finalize` tab to toggle between WF group rankings checkbox.
image
- Once you click on the Customize check box, the text fields will be activated and you can enter the values you want and then press `Update`.
- Click the `View` button to check if changes take effect.
- Click on the tool tip to get a detailed understanding how the Default and Custom values will work in the ranking algorithm.

### CODE REVIEW

File: src/edu/csus/ecs/pc2/core/model/FinalizeData.java [hunk 1]
```diff
@@ -133,4 +141,62 @@ public void setUseWFGroupRanking(boolean useWFGroupRanking) {
...
+     * @param customizeHonorsSolvedCount
+     */
+    public void setCustomizeHonorsSolvedCount(boolean customizeHonorsSolvedCount) {
+        this.customizeHonorsSolvedCount = customizeHonorsSolvedCount;
+    }
+
+    /**
+     * Return whether or not minimum problems solved for honors fields is to be customized or not
+     *
+     * @return true if WF group rankings are to be used
```
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/core/model/FinalizeData.java:157]: Javadoc for @return is wrong.  Looks like a copy/paste error.

Comment (SamanwaySadhu) [reply src/edu/csus/ecs/pc2/core/model/FinalizeData.java]: Updated!

**Review by johnbrvc — APPROVED** [review]
Requested changes made.

**Comment by SamanwaySadhu** [comment]
@johnbrvc can you assign other reviewers.

**Review by kkarakas — APPROVED** [review]
Looked at the code it looks good.

**Pull request merged**